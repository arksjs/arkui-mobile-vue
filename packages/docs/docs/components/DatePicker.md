# DatePicker/DatePickerPopup/DatePickerView 时间选择器

注：

- 这 3 个组件是渐进式组件，可查看 [渐进式组件](../design/progressive.md) 了解。

## Import

```js
import {
  TaDatePicker,
  TaDatePickerPopup,
  TaDatePickerView
} from 'tantalum-ui-mobile'
```

具体的引入方式可以参考[引入组件](../guide/import.md)。

### Import Type

组件导出的类型定义：

```ts
import type {
  DatePickerMode,
  DatePickerOptionFilter,
  DatePickerOnConfirm,
  SelectorOnChange,
  SelectorModelValue,
  DatePickerDetail,
  SelectorValueParser,
  SelectorValueFormatter,
  VisibleState,
  PopupOnVisibleStateChange,
  PopupOnCancel
} from 'tantalum-ui-mobile'
```

## 公共 Props

| 属性            | 类型                                                             | 默认值                       | 必填 | 说明                                                                                                                                                                                                                                                                                                                                 |
| --------------- | ---------------------------------------------------------------- | ---------------------------- | ---- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
| v-model         | [SelectorModelValue](./DatePicker.md#selectormodelvalue)         |                              | 否   | 选中值                                                                                                                                                                                                                                                                                                                               |
| initial-mode    | [DatePickerMode](./DatePicker.md#datepickermode)                 |                              | 否   | 选择的模式，可选 'date', 'time', 'datetime' 等                                                                                                                                                                                                                                                                                       |
| min-date        | Date                                                             | 当天对应 10 年前的 Date 实例 | 否   | 最小时间对应的 Date 实例（含）                                                                                                                                                                                                                                                                                                       |
| max-date        | Date                                                             | 当天最后一秒的 Date 实例     | 否   | 最大时间对应的 Date 实例（含）                                                                                                                                                                                                                                                                                                       |
| format-template | string                                                           |                              | 否   | 格式的模板，如 `YYYY-MM-DD`，模板规则参考 [Dayjs](https://dayjs.fenxianglu.cn/category/parse.html#%E5%AD%97%E7%AC%A6%E4%B8%B2)。设置后 v-model 的绑定变为格式化后的 string 类型，不再是 number[] 。推荐根据 `initialMode` 选型来确定，如 `initialMode='minute-second'`，则可设置 `formatTemplate='mm分ss秒'`。优先级低于 `formatter` |
| filter          | [DatePickerOptionFilter](./DatePicker.md#datepickeroptionfilter) |                              | 否   | 选项过滤器                                                                                                                                                                                                                                                                                                                           |
| formatter       | [SelectorValueFormatter](./DatePicker.md#selectorvalueformatter) |                              | 否   | 和 `parser` 成对设置，对于 v-model 和 change 的值进行转化                                                                                                                                                                                                                                                                            |
| parser          | [SelectorValueParser](./DatePicker.md#selectorvalueparser)       |                              | 否   | 和 `formatter` 成对设置，对于 v-model 和 change 的值进行反转化                                                                                                                                                                                                                                                                       |

注：v-model 对未设置的前段值采用当前时间补上，后段则采用初始时间，如：initialMode='month-day-hour'下， 选取 5 月 7 日 12 点，则 v-model 对应的时间为：`Fri May 07 2021 12:00:00 GMT+0800`，设置 `minDate`，`maxDate` 的时候需要考虑到。

## DatePicker 时间选择器

注：

- 支持表单，具体可参考 [Form](./Form.md)。

<CodeDemo name="DatePicker">

<<< @/../../demo/src/components/Form/DatePicker/index.vue

</CodeDemo>

### DatePicker Props

| 属性        | 类型    | 默认值 | 必填 | 说明                                 |
| ----------- | ------- | ------ | ---- | ------------------------------------ |
| name        | string  |        | 否   | 标识                                 |
| placeholder | string  |        | 否   | 没有选中值的提示，也会用在弹窗标题上 |
| disabled    | boolean | false  | 否   | 是否禁用                             |

### DatePicker Events

| 事件   | 描述                       | 回调函数参数                                                      | 函数 TypeScript  |
| ------ | -------------------------- | ----------------------------------------------------------------- | ---------------- |
| change | 选择后选中值发生变化时触发 | payload: [SelectorModelValue](./DatePicker.md#selectormodelvalue) | SelectorOnChange |

## DatePickerPopup 时间选择弹窗

<CodeDemo name="DatePickerPopup">

<<< @/../../demo/src/components/Form/DatePickerPopup/index.vue

</CodeDemo>

### DatePickerPopup Props

| 属性            | 类型    | 默认值 | 必填 | 说明         |
| --------------- | ------- | ------ | ---- | ------------ |
| v-model:visible | boolean | false  | 否   | 是否显示弹窗 |
| title           | string  |        | 否   | 弹窗标题     |

### DatePickerPopup Events

| 事件                 | 描述                 | 回调函数参数                                                            | 函数 TypeScript           |
| -------------------- | -------------------- | ----------------------------------------------------------------------- | ------------------------- |
| confirm              | 点击确定按钮后触发   | payload: [DatePickerDetail](./DatePicker.md#datepickerdetail)           | DatePickerOnConfirm       |
| cancel               | 点击取消按钮后触发   |                                                                         | PopupOnCancel             |
| change               | 选中值发生变化时触发 | payload: [SelectorModelValue](./DatePicker.md#selectormodelvalue)       | SelectorOnChange          |
| visible-state-change | 展示隐藏时触发       | payload: { state: [VisibleState](./DatePicker.md#visiblestate-值说明) } | PopupOnVisibleStateChange |

#### VisibleState 值说明

| 值     | 说明                 | 备注                                              |
| ------ | -------------------- | ------------------------------------------------- |
| show   | 展示时触发           |                                                   |
| shown  | 展示且动画结束后触发 |                                                   |
| hide   | 隐藏时触发           | 可能携带其他参数 cancel, maskClick, closeClick 等 |
| hidden | 隐藏且动画结束后触发 | 可能携带其他参数 cancel, maskClick, closeClick 等 |

## DatePickerView 时间选择面板

<CodeDemo name="DatePickerView">

<<< @/../../demo/src/components/Form/DatePickerView/index.vue

</CodeDemo>

### DatePickerView Events

| 事件   | 描述                       | 回调函数参数                                                      | 函数 TypeScript  |
| ------ | -------------------------- | ----------------------------------------------------------------- | ---------------- |
| change | 滑动后选中值发生变化时触发 | payload: [SelectorModelValue](./DatePicker.md#selectormodelvalue) | SelectorOnChange |

## showDatePicker(object) 显示时间选择弹窗

### object

| 属性     | 类型                                                             | 默认值                       | 必填 | 说明                                                                     |
| -------- | ---------------------------------------------------------------- | ---------------------------- | ---- | ------------------------------------------------------------------------ |
| title    | string                                                           |                              | 否   | 弹窗标题                                                                 |
| value    | number[]                                                         |                              | 否   | 默认选择值                                                               |
| mode     | [DatePickerMode](./DatePicker.md#datepickermode)                 |                              | 否   | 选择的模式，可选 'date', 'time', 'datetime' 等                           |
| minDate  | Date                                                             | 当天对应 10 年前的 Date 实例 | 否   | 最小时间对应的 Date 实例（含）                                           |
| maxDate  | Date                                                             | 当天最后一秒的 Date 实例     | 否   | 最大时间对应的 Date 实例（含）                                           |
| filter   | [DatePickerOptionFilter](./DatePicker.md#datepickeroptionfilter) |                              | 否   | 选项过滤器                                                               |
| success  | (payload: SuccessPayload) => void                                |                              | 否   | 接口调用成功（在用户做出选择后，如取消，选择选项）的回调函数             |
| fail     | (e: Error) => void                                               |                              | 否   | 接口调用失败（如传入错误的参数）的回调函数（不传入 fail 遇错误直接抛出） |
| complete | () => void                                                       |                              | 否   | 弹窗关闭或调用失败的回调函数                                             |

#### SuccessPayload

| 属性     | 类型                                                 | 说明                                          |
| -------- | ---------------------------------------------------- | --------------------------------------------- |
| confirm? | boolean                                              | 为 true 时，表示点击了确定，此时返回 `detail` |
| cancel?  | boolean                                              | 为 true 时，表示取消                          |
| detail?  | [DatePickerDetail](./DatePicker.md#datepickerdetail) |                                               |

### Usage

具体调用方式可以参考[API 调用](../guide/import.md#api-调用)。

```js
showDatePicker({
  title: 'DatePicker',
  success: ({ confirm, cancel, detail }) => {
    ...
  }
})
```

## 类型释义

### SelectorValue

```ts
type SelectorValue = string | number | Date
```

### SelectorModelValue

```ts
type SelectorModelValue = SelectorValue | SelectorValue[]
```

注：在不自定义 `formatter/parser` 的情况下，v-model 只有 `number[]` 这种情况。

### DatePickerDetail

```ts
interface DatePickerDetail {
  label: string
  value: number[]
}
```

| 字段  | 说明                                                                               |
| ----- | ---------------------------------------------------------------------------------- |
| label | 选中值对应的描述文本，如果设置了 `formatTemplate` 或 `formatter`，则返回格式后文本 |
| value | 选择的值，如 [2021, 5, 1]，不受 `formatTemplate` 或 `formatter` 影响               |

### SelectorValueFormatter

```ts
interface SelectorValueFormatter {
  (valueArray: SelectorValue[], labelArray: string[]):
    | { value: SelectorModelValue; label: string }
    | SelectorModelValue
}
```

将 v-model 的原始值转为需要的自定义值，值需要满足 `SelectorModelValue` 的类型约束，可以返回 { value, label } 对两个数据进行修改，或者单独返回 value。

### SelectorValueParser

```ts
interface SelectorValueParser {
  (value: unknown): SelectorValue[]
}
```

跟 `SelectorValueFormatter` 相反，将自定义 v-model 的值转为组件认识的原始数组。

### DatePickerMode

| 值                                | 说明                                            |
| --------------------------------- | ----------------------------------------------- |
| date                              | 日期 (同 year-month-day)                        |
| time                              | 时间 (同 hour-minute-second)                    |
| datetime                          | 日期时间 (同 year-month-day-hour-minute-second) |
| minute-second                     | 分秒                                            |
| hour-minute                       | 时分                                            |
| hour-minute-second                | 时分秒                                          |
| day-hour                          | 天时                                            |
| month-day                         | 月日                                            |
| month-day-hour                    | 月日时                                          |
| month-day-hour-minute             | 月日时分                                        |
| year-month                        | 年月                                            |
| year-month-day                    | 年月日                                          |
| year-month-day-hour               | 年月日时                                        |
| year-month-day-hour-minute        | 年月日时分                                      |
| year-month-day-hour-minute-second | 年月日时分秒                                    |

### DatePickerOptionFilter

```ts
interface DatePickerOptionFilter {
  (
    number: number,
    type: 'year' | 'month' | 'day' | 'hour' | 'minute' | 'second'
  ): boolean
}
```

通过返回 `false` 来过滤指定选项。
