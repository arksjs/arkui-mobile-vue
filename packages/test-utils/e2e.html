<script src="../../node_modules/vue/dist/vue.global.js"></script>
<script src="../../dist/index.js"></script>
<link rel="stylesheet" href="../../dist/index.css" />

<div id="app"></div>
