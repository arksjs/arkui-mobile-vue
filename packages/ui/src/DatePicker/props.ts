import type { PropType } from 'vue'
import { MODE_NAMES } from './date'
import type { Mode, OptionFilter } from './types'
import type {
  SelectorModelValue,
  SelectorValueParser,
  SelectorValueFormatter
} from '../SelectorField/types'

export const commonProps = {
  modelValue: {
    type: [Date, String, Number, Array] as PropType<SelectorModelValue>
  },
  // 格式化模板
  formatTemplate: {
    type: String
  },
  // 初始模式
  initialMode: {
    type: String as PropType<Mode>,
    default: MODE_NAMES[0]
  },
  // 最小日期
  minDate: {
    type: Date
  },
  // 最大日期
  maxDate: {
    type: Date
  },
  // 过滤器
  filter: {
    type: Function as PropType<OptionFilter>,
    default: () => true
  },
  formatter: {
    type: Function as PropType<SelectorValueFormatter>
  },
  parser: {
    type: Function as PropType<SelectorValueParser>
  }
}
