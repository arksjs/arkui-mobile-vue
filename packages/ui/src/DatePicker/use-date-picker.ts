import { provide } from 'vue'
import dayjs from '../helpers/day'
import type { Dayjs } from '../helpers/types'
import { getEnumsValue } from '../helpers/validator'
import type {
  PickerOptionsHandler,
  PickerLabelFormatter,
  PickerHandlers
} from '../Picker/types'
import type {
  SelectorValueParser,
  SelectorValueFormatter
} from '../SelectorField/types'
import {
  array2Date,
  day2Array,
  getFormatTemplate,
  MODE_NAMES,
  parseRows
} from './date'
import { rangeNumber, returnTrue } from '../helpers/util'
import type { DatePickerCommonProps } from './types'
import { getMaxDate, getMinDate, handleMinMaxDate } from './util'

export function useHandlers(props: DatePickerCommonProps) {
  const mode = getEnumsValue(MODE_NAMES, props.initialMode)
  const defaultMinDate = getMinDate()
  const defaultMaxDate = getMaxDate()

  const optionsHandler: PickerOptionsHandler = (index, parent) => {
    const { minDate, maxDate } = handleMinMaxDate(
      props.minDate ?? defaultMinDate,
      props.maxDate ?? defaultMaxDate
    )
    const filter = props.filter || returnTrue

    return parseRows(index, parent || null, {
      filter,
      minDate,
      maxDate,
      mode
    })
  }

  const parser: SelectorValueParser = value => {
    if (props.parser) {
      return props.parser(value)
    }

    let djs: Dayjs | null = null

    if (value instanceof Date) {
      djs = dayjs(value)
    } else if (
      props.formatTemplate &&
      value != null &&
      value !== '' &&
      typeof value === 'string'
    ) {
      djs = dayjs(value as string, props.formatTemplate, true)
    }

    return day2Array(djs, mode)
  }

  const defaultValueGetter = () => {
    const { minDate, maxDate } = handleMinMaxDate(
      props.minDate ?? defaultMinDate,
      props.maxDate ?? defaultMaxDate
    )

    return parser(
      new Date(rangeNumber(Date.now(), minDate.getTime(), maxDate.getTime()))
    )
  }

  const labelFormatter: PickerLabelFormatter = array => {
    return array.length === 0
      ? ''
      : dayjs(array2Date(array, mode)).format(
          getFormatTemplate(mode, props.formatTemplate)
        )
  }

  const formatter: SelectorValueFormatter = (valueArray, labelArray) => {
    if (valueArray.length === 0) {
      // 如果是空数据value用空字符串替代
      return {
        value: '',
        label: ''
      }
    }

    if (props.formatter) {
      return props.formatter(valueArray, labelArray)
    }

    const label = labelFormatter(labelArray)
    return {
      value: props.formatTemplate ? label : array2Date(valueArray, mode),
      label
    }
  }

  const handlers: PickerHandlers = {
    optionsHandler,
    formatter,
    parser,
    defaultValueGetter,
    labelFormatter
  }

  provide('akPickerHandlers', handlers)

  return { handlers }
}
