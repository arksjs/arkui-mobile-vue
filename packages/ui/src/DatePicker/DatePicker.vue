<template>
  <div class="ak-picker" :class="{ disabled }" ref="root">
    <SelectorField
      :label="fieldLabel"
      :value="fieldValue"
      :disabled="disabled"
      :name="name"
      :placeholder="placeholder"
      @fieldClick="onFieldClick"
    />
    <PickerPopup
      :title="placeholder"
      :formatter="formatter"
      :parser="parser"
      v-model:visible="popupVisible"
      v-if="isInitPopup"
      @confirm="onConfirm"
      ref="popup"
    />
  </div>
</template>

<script lang="ts">
import { defineComponent } from 'vue'
import { SelectorField } from '../SelectorField'
import { PickerPopup } from '../Picker'
import { pickerEmits, pickerProps } from '../Picker/props'
import { usePicker } from '../Picker/use-picker'
import { commonProps } from './props'
import { useHandlers } from '../DatePicker/use-date-picker'

export default defineComponent({
  name: 'ak-date-picker',
  components: { SelectorField, PickerPopup },
  props: { ...commonProps, ...pickerProps },
  emits: { ...pickerEmits },
  setup(props, ctx) {
    const { handlers } = useHandlers(props)

    return {
      ...usePicker(props, ctx, { name: 'picker', handlers })
    }
  }
})
</script>
