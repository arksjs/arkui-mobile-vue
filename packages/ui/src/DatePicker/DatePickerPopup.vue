<template>
  <Drawer
    class="fx-picker-popup fx-date-picker-popup"
    placement="bottom"
    :visible="visible"
    @visibleStateChange="onVisibleStateChange"
    @cancel="onCancel"
    @confirm="onConfirm"
    @update:visible="onUpdateVisible"
    ref="popup"
  >
    <template #header>
      <NavBar
        class="fx-drawer_header"
        :title="title"
        :leftButtons="[{ text: locale.datePickerCancelText, type: 'primary' }]"
        :rightButtons="[
          { text: locale.datePickerConfirmText, type: 'primary' }
        ]"
        :iconOnly="false"
        @leftButtonClick="onHeaderLeftClick"
        @rightButtonClick="onHeaderRightClick"
      >
      </NavBar>
    </template>
    <PickerView
      ref="viewRef"
      :modelValue="modelValue"
      :formatter="formatter"
      :parser="parser"
    />
  </Drawer>
</template>

<script lang="ts">
import { defineComponent } from 'vue'
import { PickerView } from '../Picker'
import { Drawer } from '../Drawer'
import { NavBar } from '../NavBar'
import { usePopupExtend } from '../popup/use-popup'
import { pickerPopupProps, pickerPopupEmits } from '../Picker/props'
import { usePickerPopup } from '../Picker/use-picker'
import { commonProps } from './props'
import { useHandlers } from '../DatePicker/use-date-picker'
import { useLocale } from '../ConfigProvider/context'
import type { SelectorDetail } from '../SelectorField/types'

export default defineComponent({
  name: 'fx-date-picker-popup',
  components: { PickerView, Drawer, NavBar },
  props: {
    ...pickerPopupProps,
    ...commonProps,
    title: {
      type: String,
      default: ''
    }
  },
  emits: {
    ...pickerPopupEmits
  },
  setup(props, ctx) {
    const { handlers } = useHandlers(props)
    const { locale } = useLocale()
    const popup = usePopupExtend<SelectorDetail>(ctx)
    const pickerPopup = usePickerPopup(props, ctx, popup, { handlers })

    return {
      ...popup,
      ...pickerPopup,
      locale
    }
  }
})
</script>
