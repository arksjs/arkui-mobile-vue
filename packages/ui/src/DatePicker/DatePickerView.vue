<template>
  <PickerView
    class="ta-date-picker-view"
    @update:modelValue="onUpdateValue"
    @change="onChange"
  >
  </PickerView>
</template>

<script lang="ts">
import { defineComponent } from 'vue'
import { PickerView } from '../Picker'
import { pickerViewEmits } from '../Picker/props'
import { commonProps } from './props'
import { useHandlers } from '../DatePicker/use-date-picker'
import type { SelectorModelValue } from '../SelectorField/types'
import type { PropsToEmits } from '../helpers'
import type { DatePickerViewEmits } from './types'
import type { PickerOnChange } from '../Picker/types'

export default defineComponent({
  name: 'ta-date-picker-view',
  components: { PickerView },
  props: {
    ...commonProps
  },
  emits: { ...pickerViewEmits } as PropsToEmits<DatePickerViewEmits>,
  setup(props, ctx) {
    const { emit } = ctx

    useHandlers(props)

    const onChange: PickerOnChange = (value, label) => {
      emit('change', value, label)
    }

    function onUpdateValue(e: SelectorModelValue) {
      emit('update:modelValue', e)
    }

    return {
      onChange,
      onUpdateValue
    }
  }
})
</script>
