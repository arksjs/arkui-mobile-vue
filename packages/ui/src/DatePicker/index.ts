import DatePicker from './DatePicker.vue'
import DatePickerPopup from './DatePickerPopup.vue'
import DatePickerView from './DatePickerView.vue'
import { createConfirmHook, createShowPopup } from '../popup/api'
import type { PopupSuccessConfirmArgs } from '../popup/types'
import type { SelectorDetail } from '../SelectorField/types'
import type { ShowDatePickerOptions } from './types'

const showDatePicker = createShowPopup<
  ShowDatePickerOptions,
  PopupSuccessConfirmArgs<SelectorDetail>
>({
  apiName: 'showDatePicker',
  component: DatePickerPopup,
  createHook: createConfirmHook
})

export { DatePicker, DatePickerPopup, DatePickerView, showDatePicker }
export default DatePicker
