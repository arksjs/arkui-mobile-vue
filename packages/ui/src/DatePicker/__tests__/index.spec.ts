import { describe, test, expect } from 'vitest'
import { shallowMount } from '@vue/test-utils'
import { DatePicker, DatePickerPopup, DatePickerView } from '@/DatePicker'

describe('DatePicker', () => {
  test.concurrent('snapshot', () => {
    const wrapper = shallowMount(DatePicker, {
      props: {
        options: []
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})

describe('DatePickerPopup', () => {
  test.concurrent('snapshot', () => {
    const wrapper = shallowMount(DatePickerPopup, {
      props: {
        options: []
      },
      global: {
        stubs: { teleport: true }
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})

describe('DatePickerView', () => {
  test.concurrent('snapshot', () => {
    const wrapper = shallowMount(DatePickerView, {
      props: {
        options: []
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
