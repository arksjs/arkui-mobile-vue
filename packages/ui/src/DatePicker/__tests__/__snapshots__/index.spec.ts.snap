// Vitest Snapshot v1

exports[`DatePicker > snapshot 1`] = `
"<div class=\\"ta-picker\\" options=\\"\\">
  <selector-field-stub label=\\"\\" name=\\"\\" placeholder=\\"\\" disabled=\\"false\\" value=\\"\\"></selector-field-stub>
  <!--v-if-->
</div>"
`;

exports[`DatePickerPopup > snapshot 1`] = `"<drawer-stub visible=\\"false\\" maskclosable=\\"true\\" placement=\\"bottom\\" showclose=\\"false\\" enablesafeareainsets=\\"true\\" showmask=\\"true\\" initialenableblurcancel=\\"true\\" class=\\"ta-picker-popup ta-date-picker-popup\\" options=\\"\\"></drawer-stub>"`;

exports[`DatePickerView > snapshot 1`] = `"<picker-view-stub options=\\"\\" fieldnames=\\"[object Object]\\" class=\\"ta-date-picker-view\\"></picker-view-stub>"`;
